pydantic/funcs.py

```diff
@@ -0,0 +1,541 @@
+) -> Any:
+    if __instance is _UNSET:
+        analyzed = analyze_type(__type)
+        return analyzed.serializer.to_python
+
+    serializer = _get_serializer(__type)
+    return serializer.to_python(
+        __instance,
+        **kwargs,
+    )
+
+
+class _DumpJsonCallable(Protocol, Generic[T_contra]):
+    def __call__(
+        self,
+        __instance: T_contra,
+        *,
+        indent: int | None = None,
+        include: IncEx | None = None,
+        exclude: IncEx | None = None,
+        by_alias: bool = False,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+        round_trip: bool = False,
+        warnings: bool = True,
+    ) -> bytes:
+        ...
+
+
+@overload
+def dump_json(
+    __instance: T,
+    __type: type[T],
+    *,
+    indent: int | None = ...,
+    include: IncEx | None = ...,
+    exclude: IncEx | None = ...,
+    by_alias: bool = ...,
+    exclude_unset: bool = ...,
+    exclude_defaults: bool = ...,
+    exclude_none: bool = ...,
+    round_trip: bool = ...,
+    warnings: bool = ...,
+) -> Any:
+    ...
+
+
+@overload
+def dump_json(
+    __instance: Any,
+    __type: Any,
+    *,
+    indent: int | None = ...,
+    include: IncEx | None = ...,
+    exclude: IncEx | None = ...,
+    by_alias: bool = ...,
+    exclude_unset: bool = ...,
+    exclude_defaults: bool = ...,
+    exclude_none: bool = ...,
+    round_trip: bool = ...,
+    warnings: bool = ...,
+) -> Any:
+    ...
+
+
+@overload
+def dump_json(
+    __type: type[T],
+) -> _DumpJsonCallable[T]:
+    ...
+
+
+@overload
+def dump_json(
+    __type: Any,
+) -> _DumpJsonCallable[Any]:
+    ...
+
+
+def dump_json(
+    __type: Any,
+    __instance: Any = _UNSET,
+    **kwargs: Any,
+) -> Any:
+    if __instance is _UNSET:
+        analyzed = analyze_type(__type)
+        return analyzed.serializer.to_json
+
+    serializer = _get_serializer(__type)
+    return serializer.to_json(
+        __instance,
+        **kwargs,
+    )
+
+
+def json_schema(
+    __types_or_schema: AnalyzedType[Any] | type[Any],
+    *,
+    by_alias: bool = True,
+    ref_template: str = DEFAULT_REF_TEMPLATE,
+    schema_generator: type[GenerateJsonSchema] = GenerateJsonSchema,
+) -> dict[str, Any]:
+    schema_generator_instance = schema_generator(by_alias=by_alias, ref_template=ref_template)
+
+    core_schema = (
+        __types_or_schema.core_schema
+        if isinstance(__types_or_schema, AnalyzedType)
+        else analyze_type(__types_or_schema).core_schema
+    )
+
+    return schema_generator_instance.generate(core_schema)
+
+
+def json_schemas(
+    __types_or_schemas: Iterable[AnalyzedType[Any] | type[Any]],
+    *,
+    by_alias: bool = True,
+    ref_template: str = DEFAULT_REF_TEMPLATE,
+    title: str | None = None,
+    description: str | None = None,
+    schema_generator: type[GenerateJsonSchema] = GenerateJsonSchema,
+) -> dict[str, Any]:
+    # TODO: can we use model.__schema_cache__?
```

Comment:
I removed this in one of my PRs and replaced it with a WeakKeysDict that I think was an improvement. Considering there's no weird recursion issues during schema building I think better to not have the schemas live on the models (since this also means we get improved schema caching for dataclasses etc.)